internal/pkg/agent/cmd/run.go

```diff
@@ -44,6 +47,10 @@ const (
 	agentName = "elastic-agent"
 )
 
+func init() {
+	apm.DefaultTracer.Close()
+}
```

Comment:
[Blocker]
Please move it to the tracer init function. having it here generates a side effect when importing this package that is unrelated to the package.

Reply:
@AndersonQ  I think this is debatable, I am not a fan of `init()`,  but in the case of APM you want to make sure you disable it as early in the process as we can across the project and `init` gives some guarantee of that. Metrics/traces and even logging are always problematic in the context of execution.  Especially in beats or agent, where the start-up sequence can get a bit confusing.